### PR TITLE
Add artifacts to AppVeyor build

### PR DESCRIPTION
**Pull request by MikhailTymchukDX**

### CODE REVIEW

**Comment by AlekseyMartynov** [comment]
I suggest removing the `name` attribute, `path` should be enough.
Also, can use a single wildcard: bin/AjaxControlToolkit.*

**Comment by MikhailTymchukDX** [comment]
This wildcard exposes many files:

image

Do we need all of them?

**Comment by AlekseyMartynov** [comment]
> This wildcard exposes many files

maybe `AjaxControlToolkit.???`

**Comment by AlekseyMartynov** [comment]
It appears things are more complicated. Binary artifacts should be compiled in the Release configuration, and in addition to main binaries we need HTML sanitizer dll and a ZIP with images, css and js.

It seems we need to use the AppVeyor build matrix feature. Also split bin to bin\Debug and bin\Release. It will affect the internal installation task too.

What do you think?

**Comment by MikhailTymchukDX** [comment]
Seems legit.

>a ZIP with images, css and js
Do you mean mirroring CDN.zip structure, including Bundle.js file?

**Comment by AlekseyMartynov** [comment]
I'd say mirroring the structure of the AjaxControlToolkit.StaticResources NuGet. The idea is to give users a way to easily update their projects to a nightly build.

Let's prioritize bug fixing, and this task needs design and debugging in a fork and with a temporary AppVeyor task.

File: AjaxControlToolkit/AjaxFileUpload/AjaxFileUpload.cs [hunk 1]
```diff
@@ -231,14 +231,32 @@ public bool AutoStartUpload
 
         protected override void OnInit(EventArgs e) {
             base.OnInit(e);
-            if(!IsDesignMode) {
-                if(!string.IsNullOrEmpty(Page.Request.QueryString["contextkey"])
+            if(IsDesignMode || !IsControlRequest())
```
Comment (AlekseyMartynov) [inline AjaxControlToolkit/AjaxFileUpload/AjaxFileUpload.cs]: need to merge / rebase?

Comment (MikhailTymchukDX) [inline AjaxControlToolkit/AjaxFileUpload/AjaxFileUpload.cs]: Yes, old root. Thanks for revealing it.

File: AjaxControlToolkit.Tests/JasmineTests.cs [hunk 2]
```diff
@@ -27,7 +27,7 @@ public class JasmineTests {
 
         [Test]
         public void Chrome() {
-            var dir = Path.GetDirectoryName(typeof(JasmineTests).Assembly.Location);
+            var dir = Directory.GetParent(Path.GetDirectoryName(typeof(JasmineTests).Assembly.Location)).FullName;
```
Comment (AlekseyMartynov) [inline AjaxControlToolkit.Tests/JasmineTests.cs]: This complex expression is used at least twice. Extract method.

File: AjaxControlToolkit.Tests/JasmineTests.cs [hunk 3]
```diff
@@ -108,5 +108,10 @@ public class JasmineTests {
                 Assert.Fail(exceptionMessage);
             }
         }
+
+        static string GetWebDriverDirectory() {
```
Comment (AlekseyMartynov) [inline AjaxControlToolkit.Tests/JasmineTests.cs]: Dead code?

File: AjaxControlToolkit.Jasmine/Suites/CalendarTests.aspx.cs [hunk 4]
```diff
@@ -8,16 +8,16 @@
 namespace AjaxControlToolkit.Jasmine.Suites {
     public partial class CalendarTests : System.Web.UI.Page {
         protected void Page_Load(object sender, EventArgs e) {
-            var selectedDate = new DateTime(2015, 1, 1);
```
Comment (AlekseyMartynov) [inline AjaxControlToolkit.Jasmine/Suites/CalendarTests.aspx.cs]: Unrelated changes?

File: AjaxControlToolkit/AjaxControlToolkit.csproj [hunk 5]
```diff
@@ -28,10 +28,11 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>..\bin\</OutputPath>
+    <OutputPath>..\bin\Release\</OutputPath>
     <DefineConstants>TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
+    <DocumentationFile>..\bin\AjaxControlToolkit.XML</DocumentationFile>
```
Comment (AlekseyMartynov) [inline AjaxControlToolkit/AjaxControlToolkit.csproj]: These XMLs are usually located near binaries: bin\Debug and bin\Release

File: AjaxControlToolkit/AjaxControlToolkit.csproj [hunk 6]
```diff
@@ -455,6 +456,9 @@
     <EmbeddedResource Include="Styles\NumericUpDown.min.css">
       <DependentUpon>NumericUpDown.css</DependentUpon>
     </EmbeddedResource>
+    <Content Include="..\bin\AjaxControlToolkit.XML">
```
Comment (AlekseyMartynov) [inline AjaxControlToolkit/AjaxControlToolkit.csproj]: Including from bin? Strange. Why?

**Comment by AlekseyMartynov** [comment]
Ok, I tried this in action. 
I don't like that we have 2 builds for Debug and Release, because
* It doubles the build time (obviously)
* It makes crazy things like running unit tests in Release configuration

We need a single script which will do roughly: 
* build the whole solution in Debug
* run tests
* build the main project in Release
* package Release artifacts.

**Comment by AlekseyMartynov** [comment]
Alright, finally it gets squashed!
:shipit:
